Window size config

Watchful1 opened the pull request:
Super bare bones initial implementation of passing a config value for window size into prawcore. Just wanted to have something down so I can feel like I started on it.

LilSpazJoekp commented on docs/getting_started/configuration/options.rst:
```diff
@@ -72,6 +72,7 @@ order to successfully access a third-party Reddit site:
     (default: ``t3_``).
 :subreddit_kind: The type prefix for subreddits on the :class:`.Reddit` instance
     (default: ``t5_``).
+:window_size: The number of seconds between rate limit resets (default: 600).
```
I think this should be in a different section. This section is for configuring PRAW with a custom Reddit instance.

Watchful1 replied:
Moved

LilSpazJoekp reviewed (changes requested):
Some minor changes.

LilSpazJoekp commented:
prawcore has been update. Could you rebase and squash so the pipeline can run again?

Watchful1 commented:
Still planning on finishing this, just haven't had time recently.

Watchful1 commented:
Bump, will have time for this over the holidays if not before.

Watchful1 commented:
Bump

LilSpazJoekp commented:
Just checking in on this.

Watchful1 commented:
It's still on my list, but pretty far at the bottom below other priorities. How much do you think it's actually needed? Reddit doesn't look like they are making any changes to the window size anytime soon.

LilSpazJoekp commented:
Was waiting on doing a 8.0.0 release for this. 

Would different paid tiers of rate limits make a difference in the rate limit calculation for optimal handling?

Watchful1 commented:
Extremely unlikely they would have different window sizes. They just increase the requests per window. I imagine a lot of their infrastructure is built with the 10 minute window in mind.

Watchful1 commented on docs/getting_started/logging.rst:
```diff
@@ -10,22 +10,27 @@ Add the following to your code to log everything available:
     Data: None
     Params: {'raw_json': 1}
-    Response: 200 (876 bytes)
+    Response: 200 (876 bytes) (rst-45:rem-892:used-104 ratelimit) at 1691743156.3847592
```
I did this to match the timestamp in the first line. But no real opinion whether they should both be timestamps or a human readable date.

Watchful1 commented on docs/getting_started/logging.rst:
```diff
@@ -10,22 +10,27 @@ Add the following to your code to log everything available:
-    handler.setLevel(logging.DEBUG)
+    stream_handler = logging.StreamHandler()
+    stream_handler.setLevel(logging.DEBUG)
+    file_handler = logging.handlers.RotatingFileHandler(
```
Changed

Watchful1 commented:
Addressed comments and got the build passing. I haven't touched this in a year, so I don't really remember my original thought process and whether I was still going to do anything else.

Let me know if there's any more changes you want, but no promises how fast I'll get to them.

LilSpazJoekp commented:
Woot and yeah same here it can be changed if needed later on.
